python/python/lance/torch/kmeans.py

```diff
@@ -271,6 +330,30 @@ def _transform(
         if self.metric == "cosine":
             data = torch.nn.functional.normalize(data)
 
+        if self.use_cuvs:
+            search_time_start = time.time()
+            device = torch.device("cuda")
+            out_idx = device_ndarray.empty((data.shape[0], 1), dtype="uint32")
+            out_dist = device_ndarray.empty((data.shape[0], 1), dtype="float32")
+            search_params = cagra.SearchParams(itopk_size=self.itopk_size)
+            cagra.search(
```

Comment:
is inference faster than torch significantly?

Reply:
Is this algorithm search-able on CPU? i.e., can we run indexing job on GPU but lancedb serving on CPU?

Reply:
This doesn't change the query path, so queries are still fully CPU-based.
Batch inference is significantly faster than brute-force via torch for sufficiently large numbers of centroids, at a small cost to accuracy/recall (I'll be putting up a large e2e test result before asking for re-review).